src/lib/example.js

```diff
@@ -192,28 +198,35 @@ async function extractExample(example, name, lang) {
   const spin = ora(`${step}...`).start();
 
   try {
-    const src = 'github:o1-labs/zkapp-cli#main';
-
-    // Note: Extract will overwrite any existing dir's contents. That's ok here.
-    const TEMP = '.gittar-temp-dir';
-    await gittar.extract(src, TEMP, {
-      filter(path) {
-        return path.includes(`examples/${example}/${lang}/src`);
-      },
-    });
-
-    // Example not found. Delete the project template & temp dir to clean up.
-    if (isEmpty(TEMP)) {
-      spin.fail(step);
-      console.error(_red('Example not found'));
-      sh.rm('-r', `${process.cwd()}/${name}`, TEMP);
-      return false;
+    let repoDir;
+
+    if (process.env.CI) {
+      repoDir = './';
+    } else {
+      const src = 'github:o1-labs/zkapp-cli#main';
+      const examplePath = `examples/${example}/${lang}/src`;
+
+      // Note: Extract will overwrite any existing dir's contents. That's ok here.
+      const repoDir = '.gittar-temp-dir';
+      await gittar.extract(src, repoDir, {
+        filter(path) {
+          return path.includes(examplePath);
+        },
+      });
+
+      // Example not found. Delete the project template & temp dir to clean up.
+      if (!fs.existsSync(path.join(repoDir, examplePath))) {
+        spin.fail(step);
+        console.error(_red('Example not found'));
+        sh.rm('-r', `${process.cwd()}/${name}`, repoDir);
+        return false;
+      }
     }
 
     // Delete the project template's `src` & use the example's `src` instead.
     sh.rm('-r', `${name}/src`);
-    sh.mv(`${TEMP}/examples/${example}/${lang}/src`, `${name}/src`);
-    sh.rm('-r', TEMP);
+    sh.mv(`${repoDir}/examples/${example}/${lang}/src`, `${name}/src`);
+    if (!process.env.CI) sh.rm('-r', repoDir);
```

Comment:
This is arguably a little scary if `(!process.env.CI)` is `true` when it shouldn't be, as `repoDir` would then also be `./`, meaning we'd be running `rm -r ./`. I wonder if there's a straightforward way of making this safer?

Reply:
Actually, we could emulate what `gittar.extract` does, by copying over the examples directory to a temp directory, so that in a worst case we're trying to delete a temporary directory that we named (e.g. `.gittar-temp-dir`). Maybe that's better?

Reply:
The 2nd sounds safer

Reply:
I'll go with that then! The only downside is that it'll be more code that is written to mirror the way the production code works, but isn't the production code—but that's probably worth it for safety reasons here.

Reply:
It actually ended up looking nicer also, IMO.